src/functional/merge.js

```diff
@@ -69,10 +72,19 @@ export function mergeWithSources(collection, sources, merger) {
 function deepMergerWith(merger) {
   function deepMerger(oldValue, newValue, key) {
     return isDataStructure(oldValue) && isDataStructure(newValue)
-      ? mergeWithSources(oldValue, [newValue], deepMerger)
+      ? areMergeable(oldValue, newValue)
+        ? mergeWithSources(oldValue, [newValue], deepMerger)
+        : newValue
       : merger
       ? merger(oldValue, newValue, key)
       : newValue;
   }
   return deepMerger;
 }
+
+function areMergeable(oldValue, newValue) {
```

Comment:
Yeah, no problem. I might be changing this anyway depending on what we decide on https://github.com/immutable-js/immutable-js/pull/1840#issuecomment-915296624. Should we go ahead and make it so that lists replace maps as well?

Reply:
IMO that's probably the right behavior since it is most simple and I think having a narrower behavior for the deep merge vs the top level is right.

I think we should be able to keep the top level support for merging lists-of-tuples while using the simpler replace behavior for deep conflicts.

Reply:
Done.